gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/DatasetDescriptor.java

```diff
@@ -41,4 +65,15 @@
    * This check is non-commutative.
    */
   public boolean isCompatibleWith(DatasetDescriptor other);
+
+  /**
+   * @return true if the platform-independent properties of this {@link DatasetDescriptor} are compatible with the
+   * platform-independent properties of the other {@link DatasetDescriptor}.
+   */
+  public boolean isPropertyCompatibleWith(DatasetDescriptor other);
```

Comment:
We need to clarify what is the difference between property compatibility and normal compatibility.